### PR TITLE
Contributing BSMAP

### PR DESCRIPTION
**Pull request by tloizou**
Contributing Bisulfite Sequence Mapping Program (BSMAP), a short reads mapping program for bisulfite sequencing in DNA methylation study.

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyconfigs/b/BSMAP/BSMAP-2.74-goolf-1.4.10.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+sources = [SOURCELOWER_TGZ]
+
+skipsteps = ['configure']
+
+installopts = "INSTALL_DIR=%(installdir)s"
+
+patches = [
+     'makefile.patch',
+     'param.patch',
+     'samtools.patch',
```
Comment (boegel) [inline easybuild/easyconfigs/b/BSMAP/BSMAP-2.74-goolf-1.4.10.eb]: these patch files should be renamed to include the software name and version, and give some clue of what they change/fix

e.g.

```
BSMAP-2.74_samtools-deps.patch
```

Including a line or two of comments on top of the patch file (anything before the first `---` line is comments) would be nice too, next to mentioning the author of the patch.

**Comment by tloizou** [comment]
@boegel: Changes made. Please test.

File: easybuild/easyconfigs/b/BSMAP/BSMAP-2.74_makefile-modif.patch [hunk 2]
```diff
@@ -0,0 +1,22 @@
+Patch to point to the correct bin directory in the EASYBUILDINSTALLDIR and to
+the correct location of the ZLIB libraries
+Thekla Loizou July 2015
+--- bsmap-2.74/makefile.orig	2012-09-20 23:01:57.000000000 +0300
++++ bsmap-2.74/makefile	2014-12-16 09:24:16.033631393 +0200
+@@ -1,6 +1,6 @@
+ CC=	g++
+ 
+-BIN = $(DESTDIR)/usr/bin
++BIN=$(INSTALL_DIR)/bin
```
Comment (boegel) [inline easybuild/easyconfigs/b/BSMAP/BSMAP-2.74_makefile-modif.patch]: why rename the prefix variable? can't you just use the existing DESTDIR variable? you should still drop the `/usr/` part of course, that's just stupid :)

Comment (tloizou) [inline easybuild/easyconfigs/b/BSMAP/BSMAP-2.74_makefile-modif.patch]: @boegel: If I keep the prefix variable, i.e if I have BIN=$(DESTDIR)/bin, the $DESTDIR is empty. So, it tries to install the software under /bin

Comment (boegel) [inline easybuild/easyconfigs/b/BSMAP/BSMAP-2.74_makefile-modif.patch]: well, you still need to use `installopts = "DESTDIR=%(installdir)s"` in the easyconfig files

or are you saying that `DESTDIR` is reset somehow by the Makefile itself?

if that's the case, then it's OK as-is

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/2d2cf988427d75631ac3 for a full test report.

File: easybuild/easyconfigs/b/BSMAP/BSMAP-2.74_makefile-modif.patch (hunk 2 above)
Comment (tloizou) [inline easybuild/easyconfigs/b/BSMAP/BSMAP-2.74_makefile-modif.patch]: @boegel: Yes, you are right! I don't know why I made that stupid thing :) I have fixed it now!

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/f513c12fcf8d05b8dfbb for a full test report.

**Comment by boegel** [comment]
Going in, thanks @tloizou!

**Pull request merged**